src/app/code/community/Ambimax/PriceImport/Helper/Data.php

```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+        $fileNames = $this->prepareFileNames(ftp_nlist($connectionId, $connectionInforamtions['path']));
+
+        ftp_close($connectionId);
+        //@codingStandardsIgnoreEnd
+
+        return $fileNames;
+    }
+
+    /**
+     * @return array
+     */
+    protected function _getConnectionInformations(): array
+    {
+        $connectionConfig = array(
+            'username' => $this->getFtpUser(),
+            'password' => $this->getFtpPassword(),
+            'serverAddress' => $this->getFtpServerAddress(),
+            'path' => $this->getFtpFilesLocation(),
+        );
+
+        return $connectionConfig;
+    }
+
+    protected function getFtpUser(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_username');
+
+    }
+
+    protected function getFtpPassword(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_password');
+
+    }
+
+    protected function getFtpServerAddress(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_host');
+
+    }
+
+    protected function getFtpFilesLocation(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_path');
+
+    }
+
+    protected function getFtpFileNames(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_name');
+
+    }
+
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
+    {
+        if (empty($originalFilenames)) {
+            return;
+        }
+        if (!empty($originalFilenames)) {
+            foreach ($originalFilenames as $name) {
+                $filenames[$name] = substr($name, strripos($name, '/') + 1);
+            }
+        }
+
+        return $filenames;
+    }
+
+    public function getNewestFiles(array $allFileNames): array
+    {
+        $fileName = $this->cleanFileNames($allFileNames);
+        $onlyPetfriendsFileName = $this->cleanFileNames($allFileNames, 1);
+        $fileNames = [];
+        array_push($fileNames, $fileName, $onlyPetfriendsFileName);
+        return $fileNames;
+    }
+
+    /*
+     * use $hasSuffix = 1 for list with suffix
+     */
+    public function cleanFileNames(array $fileNames, int $hasSuffix = 0): string
+    {
+        if ($hasSuffix) {
+            $fileNames = array_filter($fileNames, array($this, 'isOnlyPetfriendsFile'));
+        } else $fileNames = array_filter($fileNames, array($this, 'isPriceFile'));
+
+        sort($fileNames);
+        $newestFileName = array_pop($fileNames);
+        if (!empty($fileNames)) {
+            $this->cleanFtpFiles($fileNames);
+        }
+        return $newestFileName;
+    }
+
+    public function cleanFtpFiles(array $fileNames): void
+    {
+        foreach ($fileNames as $filename) {
+            $this->deleteFtpFiles($this->getFtpFilesLocation() . $filename);
```

Comment:
What if there is no trailing `/` in `'ambimax_priceimport/erp_import_options/file_sftp_path'`?

Reply:
then it does not work